runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/WindmillStream.java

```diff
@@ -41,6 +45,24 @@ public interface WindmillStream {
   /** Returns when the stream was opened. */
   Instant startTime();
 
+  /**
+   * Shutdown the stream. There should be no further interactions with the stream once this has been
+   * called.
+   */
+  void shutdown();
```

Comment:
clarify how this is different than close()

maybe rename close() to halfClose? that is the grpc terminology and maybe is clearer as in java closing is usually final thing

Reply:
done renamed